pkg/resources/receiver/ingress.go

```diff
@@ -0,0 +1,117 @@
+// limitations under the License.
+
+package receiver
+
+import (
+	"github.com/banzaicloud/operator-tools/pkg/reconciler"
+	netv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func (r *receiverInstance) ingressGRPC() (runtime.Object, reconciler.DesiredState, error) {
+	if r.receiverGroup.GRPCIngress != nil {
+		endpointIngress := r.receiverGroup.GRPCIngress
+		pathType := netv1.PathTypeImplementationSpecific
+		ingress := &netv1.Ingress{
+			ObjectMeta: r.receiverGroup.MetaOverrides.Merge(r.getMeta(r.receiverGroup.Name + "-grpc")),
+			Spec: netv1.IngressSpec{
+				Rules: []netv1.IngressRule{
+					{
+						Host: endpointIngress.Host,
+						IngressRuleValue: netv1.IngressRuleValue{
+							HTTP: &netv1.HTTPIngressRuleValue{
+								Paths: []netv1.HTTPIngressPath{
+									{
+										Path:     endpointIngress.Path,
+										PathType: &pathType,
+										Backend: netv1.IngressBackend{
+											Service: &netv1.IngressServiceBackend{
+												Name: r.getName(r.receiverGroup.Name),
+												Port: netv1.ServiceBackendPort{
+													Name: "grpc",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		if endpointIngress.Certificate != "" {
+			ingress.Spec.TLS = []netv1.IngressTLS{
+				{
+					Hosts:      []string{endpointIngress.Host},
+					SecretName: endpointIngress.Certificate,
+				},
+			}
+		}
+		return ingress, reconciler.StatePresent, nil
+	}
+	delete := &netv1.Ingress{
+		ObjectMeta: r.getMeta(),
+	}
+	return delete, reconciler.StateAbsent, nil
+}
+
+func (r *receiverInstance) ingressHTTP() (runtime.Object, reconciler.DesiredState, error) {
+	if r.receiverGroup.HTTPIngress != nil {
+		endpointIngress := r.receiverGroup.HTTPIngress
+		pathType := netv1.PathTypeImplementationSpecific
+		ingress := &netv1.Ingress{
+			ObjectMeta: r.receiverGroup.MetaOverrides.Merge(r.getMeta(r.receiverGroup.Name + "-remote-write")),
+			Spec: netv1.IngressSpec{
+				Rules: []netv1.IngressRule{
+					{
+						Host: endpointIngress.Host,
+						IngressRuleValue: netv1.IngressRuleValue{
+							HTTP: &netv1.HTTPIngressRuleValue{
+								Paths: []netv1.HTTPIngressPath{
+									{
+										Path:     endpointIngress.Path,
+										PathType: &pathType,
+										Backend: netv1.IngressBackend{
+											Service: &netv1.IngressServiceBackend{
+												Name: r.getName(r.receiverGroup.Name),
+												Port: netv1.ServiceBackendPort{
+													Name: "remote-write",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		if endpointIngress.Certificate != "" {
+			ingress.Spec.TLS = []netv1.IngressTLS{
+				{
+					Hosts:      []string{endpointIngress.Host},
+					SecretName: endpointIngress.Certificate,
+				},
+			}
+		}
+		return ingress, reconciler.StatePresent, nil
+	}
+	delete := &netv1.Ingress{
+		ObjectMeta: r.getMeta(),
```

Comment:
not sure, but I think the name should be identical to the one declared above
```suggestion
		ObjectMeta: r.getMeta(r.receiverGroup.Name + "-remote-write"),
```